### ISSUE TITLE
MOA Multi-label Classification Predictions

### ISSUE THREAD

**Issue by AdeboyeML** [broadinstitute/lincs-profiling-comparison#8]
MOA Multi-label Classification Predictions 

@gway @shntnu 

- A compound profile has multiple MOAs i.e. more than one target label (multi-label classification)
-  All MOAs found in just one compound are given the same MOA label called  **unknown moa** 
- Train / Test split == ~78% / ~22%
- 271 MOAs labels
- Level 4 profiles

![image](https://user-images.githubusercontent.com/56560275/108089693-e4e2f480-7047-11eb-8eb9-f5a7557c9f24.png)

### - Above figure interpretation

- ~ 187 MOAs are found in 2 - 5 compounds
- ~ 67 MOAs are found in 6 - 15 compounds
- ~ 10 MOAs are found in 16 - 25 compounds
- ~ 5 MOAs are found in 26 - 42 compounds
- 1 MOA (unknown moa) is found in 248 compounds

### Overall Model Prediction results

![image](https://user-images.githubusercontent.com/56560275/108096208-b3b9f280-704e-11eb-9d63-7106a58bb238.png)

### - Model predictions with respect to MOA distribution among the 1,398 distinct compounds

![image](https://user-images.githubusercontent.com/56560275/108096457-fe3b6f00-704e-11eb-8af1-b2b8d8112de9.png)

#### - Cell painting
![image](https://user-images.githubusercontent.com/56560275/108096693-465a9180-704f-11eb-9d7d-bf17881ab0a7.png)

#### - L1000
![image](https://user-images.githubusercontent.com/56560275/108096918-802b9800-704f-11eb-8ace-bd94f6ec9571.png)

#### - Cell painting & L1000
![image](https://user-images.githubusercontent.com/56560275/108097008-9afe0c80-704f-11eb-9fc2-be9154b307d3.png)

### 30 different MOAs with higher ROC-AUC score in one profiling assay than in another one

![image](https://user-images.githubusercontent.com/56560275/108097173-cd0f6e80-704f-11eb-801c-65040dfbee1a.png)

![image](https://user-images.githubusercontent.com/56560275/108101113-cf27fc00-7054-11eb-840d-28c0388c7adf.png)

#### - No MOA have higher ROC-AUC score in L1000 than in Cell painting and Integrated Cell painting & L1000 level 4 data

**Comment by shntnu** [comment]
Thanks @AdeboyeML 

- Please clarify how you did the train test split – was it 80/20 per MOA? (which then works out to be ~78/22) What do you do with MOAs with few compounds? How do you handle compounds with multiple MOAs
- Please plot a graph similar to the first one, reporting the "polypharmacology" i.e. number of MOAs per compound (X = number of MOAs of that a compound is annotated with; Y = number of compounds)
- How do you assign a label? Presumably not a max because that would give you only one label.

We can discuss over a call if that's easier

**Comment by AdeboyeML** [comment]
@shntnu Yes we can clarify some of the above bullet points over a zoom call.

**Comment by AdeboyeML** [comment]
@shntnu @gwaygenomics 

### - Polypharmacology


![image](https://user-images.githubusercontent.com/56560275/108279183-2f4c9a00-714a-11eb-9d8e-230a77bb25fd.png)
